woocommerce/woocommerce-ios
# Coupon Management: Merge feature branch to trunk

### Description

This PR merges feature branch feat/coupon-management to `trunk` to continue implementation for Coupon Management project and avoid future conflicts. Changes include all of the following, which can be reviewed individually through commit history:
- All of @joshheald's work during his trial
- Updates of generated files for latest models
- New data model version 60 with new entity Coupon
- Feature flag for coupon management
- New option on Settings screen to navigate to Coupon Management screen under the feature flag (which should be updated later if we want to move Coupon Management to More menu).

### Testing instructions

- Make sure that CI passes and no warnings are found when building the project in Xcode.
- Make sure that app builds and runs successfully without crashing.
- Test Coupon Management screen:
   - Log in to a valid account.
   - On My Store tab, navigate to Settings screen.
   - Notice that there's an option for Manage Coupons under Store Settings option.
   - Tap on Manage Coupons, notice that you are navigated to Coupons screen.
   - Turn off feature flag for coupon management, notice that Manage Coupons menu is no longer available on Settings screen.

### Screenshots

https://user-images.githubusercontent.com/5533851/147344192-14f16b13-e776-47b3-aad1-f5e2619c2841.mp4


---
